addons/refuel/CfgVehicles.hpp

```diff
@@ -0,0 +1,346 @@
+                    exceptions[] = {"isNotInside"}; \
+                    icon = PATHTOF(ui\icon_refuel_interact.paa); \
+                }; \
+            }; \
+        }; 
+#define MACRO_CONNECT_ACTIONS \
+        class ACE_Actions { \
+            class ACE_MainActions { \
+                class GVAR(connect) { \
+                    displayName = CSTRING(connect); \
+                    distance = REFUEL_ACTION_DISTANCE; \
+                    condition = QUOTE([ARR_2(_player,_target)] call FUNC(canConnectNozzle)); \
+                    statement = QUOTE([ARR_2(_player,_target)] call DFUNC(connectNozzle)); \
+                    exceptions[] = {"isNotInside"}; \
+                    icon = PATHTOF(ui\icon_refuel_interact.paa); \
+                }; \
+                class GVAR(Disconnect) { \
+                    displayName = CSTRING(Disconnect); \
+                    distance = REFUEL_ACTION_DISTANCE; \
+                    condition = QUOTE([ARR_2(_player,_target)] call FUNC(canDisconnect)); \
+                    statement = QUOTE([ARR_2(_player,_target)] call DFUNC(disconnect)); \
+                    exceptions[] = {"isNotInside"}; \
+                    icon = PATHTOF(ui\icon_refuel_interact.paa); \
+                }; \
+                class GVAR(TurnOff) { \
+                    displayName = CSTRING(TurnOff); \
+                    distance = REFUEL_ACTION_DISTANCE; \
+                    condition = QUOTE([ARR_2(_player,_target)] call FUNC(canTurnOff)); \
+                    statement = QUOTE([ARR_2(_player,_target)] call DFUNC(turnOff)); \
+                    exceptions[] = {"isNotInside"}; \
+                    icon = PATHTOF(ui\icon_refuel_interact.paa); \
+                }; \
+            }; \
+        }; 
+
+// does not work
+#define MACRO_NOZZLE_ACTIONS \
+        class ACE_Actions { \
+            class ACE_MainActions { \
+                displayName = ECSTRING(interaction,MainAction); \
+                selection = ""; \
+                distance = 2; \
+                condition = "true"; \
+                class GVAR(TakeNozzle) { \
+                    displayName = CSTRING(TakeNozzle); \
+                    distance = REFUEL_ACTION_DISTANCE; \
+                    condition = QUOTE([ARR_2(_player,_target)] call FUNC(canTakeNozzle)); \
+                    statement = QUOTE([ARR_3(_player,_target,_target)] call FUNC(TakeNozzle)); \
+                    exceptions[] = {"isNotInside"}; \
+                    icon = PATHTOF(ui\icon_refuel_interact.paa); \
+                }; \
+            }; \
+        };
+        
+class CfgVehicles {
+    class ACE_Module;
+    class ACE_moduleRefuelSettings : ACE_Module {
+        scope = 2;
+        displayName = CSTRING(RefuelSettings_Module_DisplayName);
+        icon = QUOTE(PATHTOF(ui\icon_module_refuel.paa));
+        category = "ACE";
+        function = QUOTE(DFUNC(moduleRefuelSettings));
+        functionPriority = 1;
+        isGlobal = 1;
```

Comment:
Settings modules do not have to be global, as they are only relevant on the server side.